### ISSUE TITLE
parameter `ignore_keys` of `trainer.predict` not accessible in `Trainer` or `TrainingArguments`

### ISSUE THREAD

**Issue by shabie** [huggingface/transformers#11719]
# 🚀 Feature request

The [`predict`](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.predict) and [`evaluate`](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.evaluate) methods of the Trainer class provide an excellent option of `ignore_keys`. Here is a small example:

```python
trainer.predict(dataset, ignore_keys=["ner_loss", "cls_loss", "ner_logits", "cls_logits"])
```

@sgugger 

This option is however, not accessible during the normal setup of defining `TrainingArguments` class nor the `Trainer` class so the a call to `trainer.train()` leads to errors during the mid-training evaluation.

## Motivation

I am unable to evaluate the model metrics on the validation set **during** the training to see if makes sense to continue.

## Your contribution

I am happy to make a PR if this is seen as a genuine problem. Like always, may be I am missing something.

**Comment by sgugger** [comment]
We could add an argument for this (like `ignore_keys_for_eval`) yes. Let me know if you want to tackle this!

**Comment by shabie** [comment]
I very much wanna do this and will get right down to it!  👍 Admittedly though, I am relatively new to making contributions.

**Comment by shabie** [comment]
Hi! As a short update: I will not be able to work on this until 23rd of June... so if anyone wants to pick it up good otherwise I need 3 more weeks.

**Comment by shabie** [comment]
OK so I'd like to work on this :)

**Comment by sgugger** [comment]
Sure! Ping me when you open a PR :-)
